### ISSUE TITLE
 How to write my own modules when engine=slimerjs,it doesn't work

### ISSUE THREAD

**Issue by flyHe** [casperjs/casperjs#1293]
How to write my own modules when engine=slimerjs,it doesn't work,Casper will hang at require line.But it works when engine=phantomjs,How to fix it?Code is below:

``` javascript
// my module, stored in services.js
// patching phantomjs' require()
var require = patchRequire(require);
// now you're ready to go
/*[testHomePage description]*/
exports.testHomePage = function(test) {
    //work
};
```

//main script

``` javascript
var services = require('services');
```

// run with command
$ casperjs test test.js --engine=slimerjs

**Comment by flyHe** [comment]
Well,I find the same issue which has been solved
[GitHub](https://github.com/n1k0/casperjs/issues/1229)

**Comment by flyHe** [comment]
Well,I find the same issue which has been solved
https://github.com/n1k0/casperjs/issues/1229!
[Cannot require local scripts while using the slimerjs engine](https://github.com/n1k0/casperjs/issues/1229)

**Comment by flyHe** [comment]
Well,I find the same issue https://github.com/n1k0/casperjs/issues/846

**Comment by flyHe** [comment]
Slimerjs(0.9.6) relative require sitll not works,But when I used the absolute path,it worked.
```javascript
var fs = require('fs');
var path = fs.absolute(fs.workingDirectory + '/services');
var services = require(path);
```

**Comment by flyHe** [comment]
```javascript
// my module, stored in services.js
// patching phantomjs' require()
//var require = patchRequire(require);

// now you're ready to go
/**
 * [testHomePage description]
 * @param  {[type]} test [description]
 * @return {[type]}      [description]
 */
exports.testHomePage = function(test) {
	console.log("End");
	casper.start();
	casper.echo(new Date() + " Start to test 'homepage'", "info");
        casper.open(Url);
	casper.echo(new Date() + " 'Homepage' is opening", "info");
        work...
};
```
It's services.js code.Run phantomjs--it worked;run slimerjs--it apeared an error
###FAIL referenceError: casper is not define

**Comment by flyHe** [comment]
When the service.js code is below:
```javascript
// my module, stored in services.js
// patching phantomjs' require()
//var require = patchRequire(require);

// now you're ready to go
/**
 * [testHomePage description]
 * @param  {[type]} test [description]
 * @return {[type]}      [description]
 */
exports.testHomePage = function(test) {
    console.log("End");
};
```
It appeared an error:
<h3>FAIL TypeError: 'caller', 'callee',and 'arguments' properties may not be accessed on strict mode functions or the arguments objects for calls to them

**Comment by christianhaller** [comment]
+1

**Comment by istr** [comment]
@christianhaller Could you please indicate the CasperJS version you're using? Is it using a packaged version or using master?

**Comment by christianhaller** [comment]
@istr my required module needs casper. I'm now injecting it :) Problem solved (1.1.0 beta3)

**Comment by kensoh** [comment]
Closing the issue for now as it seems to have a fix from @christianhaller but do reopen if otherwise!
